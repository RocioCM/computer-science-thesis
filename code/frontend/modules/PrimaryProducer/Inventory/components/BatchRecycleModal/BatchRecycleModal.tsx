import Button from '@/common/components/Button';
import Modal, { ModalProps } from '@/common/components/Modal';
import { toast } from 'react-toastify';
import PrimaryProducerServices from '../../services';
import { RecyclePayload } from '../../types';
import useForm from '@/common/hooks/useForm';
import { RECYCLE_FORM_INPUTS, RECYCLE_FORM_STRUCT } from '../../constants';
import { HTTP_STATUS } from '@/common/constants';

interface Props extends ModalProps {
  handleCancel: () => any;
  editingId: number | null;
  handleSuccess: () => void;
}

const BatchRecycleModal: React.FC<Props> = ({
  editingId,
  handleCancel,
  handleSuccess,
  ...props
}) => {
  const { form, submitEnabled, formBuilder } = useForm(RECYCLE_FORM_STRUCT);

  const handleRecycle = async () => {
    const payload: RecyclePayload = {
      batchId: editingId as number,
      quantity: form.quantity,
    };
    const { ok, status } = await PrimaryProducerServices.recycleBatch(payload);
    if (ok) {
      toast.success('Envases enviados para reciclaje correctamente');
      handleCancel();
      handleSuccess();
    } else {
      if (status === HTTP_STATUS.badRequest) {
        toast.error(
          'No puedes reciclar más productos de los que tienes disponibles en tu lote'
        );
      } else {
        toast.error(
          'No se pudo completar el retiro de circulación. Por favor, inténtelo de nuevo más tarde'
        );
      }
    }
  };

  return (
    <Modal handleCancel={handleCancel} {...props}>
      <h2 className="w-full font-medium text-[1.75rem]  bg-n0 border-b border-n2 pb-2 mb-1">
        Reciclaje de lote #{editingId}
      </h2>
      <div className="w-full flex flex-col gap-m my-xl">
        {formBuilder(RECYCLE_FORM_INPUTS)}
      </div>
      <Button
        handleClick={handleRecycle}
        disabled={!submitEnabled}
        width="full"
        label="Enviar para reciclaje"
      />
    </Modal>
  );
};

export default BatchRecycleModal;
