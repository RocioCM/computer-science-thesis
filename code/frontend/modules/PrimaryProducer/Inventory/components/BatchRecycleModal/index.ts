import BatchRecycleModal from './BatchRecycleModal';

export default BatchRecycleModal;
