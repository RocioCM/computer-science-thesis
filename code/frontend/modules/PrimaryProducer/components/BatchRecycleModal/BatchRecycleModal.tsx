import Modal, { ModalProps } from '@/common/components/Modal';

interface Props extends ModalProps {
  handleCancel: () => any;
  editingId: number | null;
  handleSuccess: () => void;
}

const BatchRecycleModal: React.FC<Props> = ({
  editingId,
  handleCancel,
  handleSuccess,
  ...props
}) => {
  return (
    <Modal handleCancel={handleCancel} big skippable={false} {...props}>
      <h2 className="w-full bg-n0">Reciclar lote de botellas #{editingId}</h2>
    </Modal>
  );
};

export default BatchRecycleModal;
