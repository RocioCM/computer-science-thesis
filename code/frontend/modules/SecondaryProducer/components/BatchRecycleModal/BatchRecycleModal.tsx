import Button from '@/common/components/Button';
import Modal, { ModalProps } from '@/common/components/Modal';
import { toast } from 'react-toastify';
import SecondaryProducerServices from '../../services';
import { RecycleBaseBottlesPayload } from '../../types';
import useForm from '@/common/hooks/useForm';
import { RECYCLE_FORM_INPUTS, RECYCLE_FORM_STRUCT } from '../../constants';

interface Props extends ModalProps {
  handleCancel: () => any;
  editingId: number | null;
  handleSuccess: () => void;
}

const BatchRecycleModal: React.FC<Props> = ({
  editingId,
  handleCancel,
  handleSuccess,
  ...props
}) => {
  const { form, submitEnabled, formBuilder } = useForm(RECYCLE_FORM_STRUCT);

  const handleRecycle = async () => {
    const payload: RecycleBaseBottlesPayload = {
      productBatchId: editingId as number,
      quantity: form.quantity,
    };
    const { ok } = await SecondaryProducerServices.recycleBatch(payload);
    if (ok) {
      toast.success('Envases enviados para reciclaje correctamente');
      handleCancel();
      handleSuccess();
    } else {
      toast.error(
        'No se pudo completar el retiro de circulación. Por favor, inténtelo de nuevo más tarde'
      );
    }
  };

  return (
    <Modal handleCancel={handleCancel} {...props}>
      <h2 className="w-full bg-n0">Reciclaje de lote #{editingId}</h2>
      <div className="w-full flex flex-col gap-m my-xl">
        {formBuilder(RECYCLE_FORM_INPUTS)}
      </div>
      <Button
        handleClick={handleRecycle}
        disabled={!submitEnabled}
        width="full"
        label="Enviar para reciclaje"
      />
    </Modal>
  );
};

export default BatchRecycleModal;
