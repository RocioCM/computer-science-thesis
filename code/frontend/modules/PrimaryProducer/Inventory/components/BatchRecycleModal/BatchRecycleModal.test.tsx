import { render } from '@testing-library/react';
import BatchRecycleModal from './BatchRecycleModal';

describe('Primary Producer Inventory - BatchRecycleModal', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Matches snapshot', () => {
    const { container } = render(
      <div id="modal-root">
        <BatchRecycleModal
          editingId={1}
          handleCancel={() => {}}
          handleSuccess={() => {}}
        />
      </div>
    );
    expect(container).toMatchSnapshot();
  });
});
