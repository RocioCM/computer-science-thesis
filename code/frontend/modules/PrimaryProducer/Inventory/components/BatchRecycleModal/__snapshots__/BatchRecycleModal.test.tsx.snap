// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Producer Inventory - BatchRecycleModal Matches snapshot 1`] = `<div />`;
